_test/test_mpi/test_ParpoolMPI_Framework.m

```diff
@@ -15,24 +16,25 @@
             this = this@MPI_Test_Common(name);
         end
         %
-        function test_probe_all_receive_all(this,varargin)
+        function test_probe_all_receive_all(obj,varargin)
             % common code -------------------------------------------------
-            if this.ignore_test
+            if obj.ignore_test
                 return;
             end
             if nargin>1
-                this.setUp();
-                clob0 = onCleanup(@()tearDown(this));
+                obj.setUp();
+                clob0 = onCleanup(@()tearDown(obj));
             end
             
             cl = parcluster();
             num_labs = cl.NumWorkers;
             if num_labs < 3
+                warning('Can not run test_send_receive_message, not enough workers');
                 return;
             end
             num_labs = 3*floor(num_labs/3);
-            if num_labs > 27
-                num_labs = 27;
+            if num_labs > 6
```

Comment:
why has this dropped from `num_labs > 27` to `> 6`?

Reply:
it may be a problem on Windows machine, when two jobs are running simultaneously. It could run 21 there, but is selecting 9, and two jobs may overload memory and processors. I do not remember the idea of the test, but 3 are certanly necessary to check all different types of communications and may be sufficent after careful thinking. I do not want to think about it so put 6